src/native/managed/cdac/Microsoft.Diagnostics.DataContractReader.Contracts/Contracts/ExecutionManager/ExecutionManagerCore.ReadyToRunJitManager.cs

```diff
@@ -97,29 +63,127 @@ public override bool GetMethodInfo(RangeSection rangeSection, TargetCodePointer
+                < 11 => 3,
+                >= 11 => 4,
+            };
+        }
+
+        private uint GetUnwindDataSize()
+        {
+            RuntimeInfoArchitecture arch = Target.Contracts.RuntimeInfo.GetTargetArchitecture();
+            return arch switch
+            {
+                RuntimeInfoArchitecture.X86 => sizeof(uint),
+                _ => throw new NotSupportedException($"GetUnwindDataSize not supported for architecture: {arch}")
+            };
+        }
+
+        #region RuntimeFunction Helpers
+
+        private Data.ReadyToRunInfo GetReadyToRunInfo(RangeSection rangeSection)
+        {
             if (rangeSection.Data == null)
                 throw new ArgumentException(nameof(rangeSection));
 
             Debug.Assert(rangeSection.Data.R2RModule != TargetPointer.Null);
 
             Data.Module r2rModule = Target.ProcessedData.GetOrAdd<Data.Module>(rangeSection.Data.R2RModule);
             Debug.Assert(r2rModule.ReadyToRunInfo != TargetPointer.Null);
-            Data.ReadyToRunInfo r2rInfo = Target.ProcessedData.GetOrAdd<Data.ReadyToRunInfo>(r2rModule.ReadyToRunInfo);
+            return Target.ProcessedData.GetOrAdd<Data.ReadyToRunInfo>(r2rModule.ReadyToRunInfo);
+        }
+
+        private bool GetRuntimeFunction(
+            RangeSection rangeSection,
+            Data.ReadyToRunInfo r2rInfo,
+            TargetCodePointer jittedCodeAddress,
+            out TargetPointer imageBase,
+            out uint runtimeFunctionIndex)
+        {
+            imageBase = TargetPointer.Null;
+            runtimeFunctionIndex = 0;
+
+            if (rangeSection.Data == null)
+                throw new ArgumentException(nameof(rangeSection));
 
             // Check if address is in a thunk
             if (IsStubCodeBlockThunk(rangeSection.Data, r2rInfo, jittedCodeAddress))
-                return TargetPointer.Null;
+                return false;
 
             // Find the relative address that we are looking for
             TargetPointer addr = CodePointerUtils.AddressFromCodePointer(jittedCodeAddress, Target);
-            TargetPointer imageBase = rangeSection.Data.RangeBegin;
+            imageBase = rangeSection.Data.RangeBegin;
             TargetPointer relativeAddr = addr - imageBase;
 
-            uint index;
-            if (!_runtimeFunctions.TryGetRuntimeFunctionIndexForAddress(r2rInfo.RuntimeFunctions, r2rInfo.NumRuntimeFunctions, relativeAddr, out index))
-                return TargetPointer.Null;
+            return _runtimeFunctions.TryGetRuntimeFunctionIndexForAddress(r2rInfo.RuntimeFunctions, r2rInfo.NumRuntimeFunctions, relativeAddr, out runtimeFunctionIndex);
+        }
 
-            return _runtimeFunctions.GetRuntimeFunctionAddress(r2rInfo.RuntimeFunctions, index);
+        private uint AdjustRuntimeFunctionIndexForHotCold(Data.ReadyToRunInfo r2rInfo, uint index)
+        {
+            bool featureEHFunclets = Target.ReadGlobal<byte>(Constants.Globals.FeatureEHFunclets) != 0;
```

Comment:
Nit: I do not think we need to bother maintaining FeatureEHFunclets checks. It should be fine to assume funclets for cdac.